packages/core/src/debugr/debugr.ts

```diff
@@ -0,0 +1,143 @@
+import EventEmitter from 'events';
+import { EventDispatcher, Events } from '../events';
+import {
+  FormatterPlugin,
+  isFormatterPlugin,
+  Plugin,
+  PluginId,
+  PluginManager,
+  Plugins,
+} from '../plugins';
+import { Logger, LogHandler, TContextBase } from '../logger';
+
+export class Debugr<
+  TContext extends TContextBase = { processId: string },
+  TGlobalContext extends Record<string, any> = {},
+> {
+  private readonly eventDispatcher: EventDispatcher;
+
+  public readonly pluginManager: PluginManager<Partial<TContext>, TGlobalContext>;
+
+  private readonly loggerInternal: Logger<Partial<TContext>, TGlobalContext>;
+
+  private constructor(
+    eventDispatcher: EventDispatcher,
+    pluginManager: PluginManager<Partial<TContext>, TGlobalContext>,
+    logger: Logger<Partial<TContext>, TGlobalContext>,
+    plugins: Plugin<Partial<TContext>, TGlobalContext>[],
+  ) {
+    this.eventDispatcher = eventDispatcher;
+    this.pluginManager = pluginManager;
+    this.loggerInternal = logger;
+
+    for (const plugin of plugins) {
+      this.pluginManager.register(plugin);
+    }
+  }
+
+  public get logger(): Logger<Partial<TContext>, TGlobalContext> | never {
+    this.checkFormatters();
```

Comment:
tohle bych dělal na konci konstruktoru, jinak to vyhodí případnou chybu na dost non-obvious místě (bylo by potřeba zrušit `register*()` metody, ale to asi nevadí

Reply:
done